pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -502,6 +539,8 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 							Labels:           labels,
 							AvailabilityZone: az,
 							ServiceAccount:   sa,
+							ReadinessProbes:  readiness,
```

Comment:
make sure that readiness and liveness are only attached to `ServiceInstance` if `port.Port` matches httpPort otherwise it is not a probe associated with this instance.